### PR TITLE
fix: Voice Chat Echo Cancellation workaround

### PR DESCRIPTION
**Pull request by pablitar**
See here: https://bugs.chromium.org/p/chromium/issues/detail?id=687574

Also:
- Added an action to toggle recording instead of push to talk (with Alt + T)
- Reduced buffers to decrease latency

### CODE REVIEW

File: kernel/packages/unity-interface/BrowserInterface.ts [hunk 1]
```diff
@@ -376,8 +380,9 @@ export class BrowserInterface {
   }
 
   async RequestGIFProcessor(data: { imageSource: string; id: string; isWebGL1: boolean }) {
-    // tslint:disable-next-line
-    const isSupported = typeof OffscreenCanvas !== 'undefined' && typeof OffscreenCanvasRenderingContext2D === 'function'
+    const isSupported =
```
Comment (nchamo) [inline kernel/packages/unity-interface/BrowserInterface.ts:383]: This looks kind of weird

File: unity-client/ProjectSettings/QualitySettings.asset [hunk 2]
```diff
@@ -21,7 +21,7 @@ QualitySettings:
     skinWeights: 4
     textureQuality: 0
     anisotropicTextures: 2
-    antiAliasing: 2
+    antiAliasing: 0
```
Comment (nchamo) [inline unity-client/ProjectSettings/QualitySettings.asset]: Was this intended?

File: kernel/packages/voice-chat-codec/VoiceCommunicator.ts [hunk 3]
```diff
@@ -82,8 +85,17 @@ export class VoiceCommunicator {
     this.channelBufferSize = this.options.channelBufferSize ?? 2.0
 
     this.context = new AudioContext({ sampleRate: this.sampleRate })
+
     this.outputGainNode = this.context.createGain()
-    this.outputGainNode.connect(this.context.destination)
+
+    if (options.loopbackAudioElement) {
+      // Workaround for echo cancellation. See: https://bugs.chromium.org/p/chromium/issues/detail?id=687574#c71
```
Comment (nchamo) [inline kernel/packages/voice-chat-codec/VoiceCommunicator.ts:92]: Chrooooome!!!! 😡

**Review by nchamo — APPROVED** [review]
Great work!

File: unity-client/ProjectSettings/QualitySettings.asset (hunk 2 above)
Comment (pablitar) [reply unity-client/ProjectSettings/QualitySettings.asset]: Nope. Will revert this.

File: kernel/packages/unity-interface/BrowserInterface.ts (hunk 1 above)
Comment (pablitar) [reply kernel/packages/unity-interface/BrowserInterface.ts:383]: It does. But is a combination of two factors:
- The formatter which will change this line every time I save
- The linter which is wrong.

I decided to give in and make the modification so it wouldn't reformat the file anymore haha.

File: kernel/packages/voice-chat-codec/VoiceCommunicator.ts (hunk 3 above)
Comment (pablitar) [reply kernel/packages/voice-chat-codec/VoiceCommunicator.ts:92]: image

**Pull request merged**